candle-transformers/src/models/bert.rs

```diff
@@ -232,8 +250,8 @@ impl BertSelfAttention {
     }
 }
 
-impl Module for BertSelfAttention {
-    fn forward(&self, hidden_states: &Tensor) -> Result<Tensor> {
+impl BertSelfAttention {
+    fn forward(&self, hidden_states: &Tensor, attention_mask: &Option<Tensor>) -> Result<Tensor> {
```

Comment:
What is the best approach to add this extra parameter? I fear removing the trait impl is not a good idea?